packages/app/src/metadata.ts

```diff
@@ -1,440 +1,5 @@
-          clickhouse_settings: {
-            max_rows_to_read: DEFAULT_SAMPLE_SIZE,
-            read_overflow_mode: 'break',
-          },
-        })
-          .then(res => res.json<Record<string, unknown>>())
-          .then(d => d.data.map(row => row.value as string));
-        return values;
-      },
-    );
-  }
-
-  async getAllFields({
-    databaseName,
-    tableName,
-    connectionId,
-  }: {
-    databaseName: string;
-    tableName: string;
-    connectionId: string;
-  }) {
-    const fields: Field[] = [];
-    const columns = await this.getColumns({
-      databaseName,
-      tableName,
-      connectionId,
-    });
-
-    for (const c of columns) {
-      fields.push({
-        path: [c.name],
-        type: c.type,
-        jsType: convertCHDataTypeToJSType(c.type),
-      });
-    }
-
-    const mapColumns = filterColumnMetaByType(columns, [JSDataType.Map]) ?? [];
-
-    await Promise.all(
-      mapColumns.map(async column => {
-        const keys = await this.getMapKeys({
-          databaseName,
-          tableName,
-          column: column.name,
-          connectionId,
-        });
-
-        const match = column.type.match(/Map\(.+,\s*(.+)\)/);
-        const chType = match?.[1] ?? 'String'; // default to string ?
-
-        for (const key of keys) {
-          fields.push({
-            path: [column.name, key],
-            type: chType,
-            jsType: convertCHDataTypeToJSType(chType),
-          });
-        }
-      }),
-    );
-
-    return fields;
-  }
-
-  async getTableMetadata({
-    databaseName,
-    tableName,
-    connectionId,
-  }: {
-    databaseName: string;
-    tableName: string;
-    connectionId: string;
-  }) {
-    const tableMetadata = await Metadata.queryTableMetadata({
-      cache: this.cache,
-      database: databaseName,
-      table: tableName,
-      connectionId,
-    });
-
-    return tableMetadata;
-  }
-
-  async getKeyValues({
-    chartConfig,
-    keys,
-    limit = 20,
-  }: {
-    chartConfig: ChartConfigWithDateRange;
-    keys: string[];
-    limit?: number;
-  }) {
-    const sql = await renderChartConfig({
-      ...chartConfig,
-      select: keys
-        .map((k, i) => `groupUniqArray(${limit})(${k}) AS param${i}`)
-        .join(', '),
-    });
-
-    const json = await sendQuery<'JSON'>({
-      query: sql.sql,
-      query_params: sql.params,
-      connectionId: chartConfig.connection,
-      clickhouse_settings: {
-        max_rows_to_read: DEFAULT_SAMPLE_SIZE,
-        read_overflow_mode: 'break',
-      },
-    }).then(res => res.json<any>());
-
-    return Object.entries(json.data[0]).map(([key, value]) => ({
-      key: keys[parseInt(key.replace('param', ''))],
-      value: (value as string[])?.filter(Boolean), // remove nulls
-    }));
-  }
-}
-
-export type Field = {
-  path: string[];
-  type: string;
-  jsType: JSDataType | null;
-};
-
-export const metadata = new Metadata();
+export const getMetadata = () => _getMetadata(getClickhouseClient());
```

Comment:
why did we change this from a singleton? aren't we creating a ton of new metadata objects without reason?